File: SD/src/main/java/org/example/Cliente.java
package org.example;

import java.io.*;
import java.net.*;
import org.json.JSONObject;

public class Cliente {
    public static void main(String[] args) {
        String serverAddress = "127.0.0.1";
        int serverPort = 12345;

        try {
            Socket socket = new Socket(serverAddress, serverPort);

            OutputStream outputStream = socket.getOutputStream();
            PrintWriter out = new PrintWriter(outputStream, true);

            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

            for (int i = 0; i < 5; i++) {
                System.out.println("Digite os dados (Nome, CPF, idade, mensagem) na seguinte forma:");

                String inputData = reader.readLine();
                String[] dataParts = inputData.split(",");

                if (dataParts.length != 4) {
                    System.out.println("Formato incorreto. Certifique-se de inserir Nome, CPF, idade e mensagem separados por vírgula.");
                    continue;
                }

                String nome = dataParts[0].trim();
                String cpf = dataParts[1].trim();
                int idade = 0;
                String mensagem = "";

                try {
                    idade = Integer.parseInt(dataParts[2].trim());
                } catch (NumberFormatException e) {
                    System.out.println("Idade deve ser um número inteiro.");
                    continue;
                }

                mensagem = dataParts[3].trim();

                JSONObject jsonMessage = new JSONObject();
                jsonMessage.put("Nome", nome);
                jsonMessage.put("CPF", cpf);
                jsonMessage.put("idade", idade);
                jsonMessage.put("mensagem", mensagem);

                out.println(jsonMessage.toString());
            }

            socket.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: Sockets/Cliente.py
import socket
import json
from builtins import print


def main():
    server_address = "127.0.0.1"
    server_port = 8080  # Use port 8080

    try:
        with socket.create_connection((server_address, server_port)) as client_socket:
            for i in range(5):
                print("Digite os dados (Nome, CPF, idade, mensagem) na seguinte forma:")
                input_data = input()
                data_parts = input_data.split(",")
                print()
                if len(data_parts) != 4:
                    print("Formato incorreto. Certifique-se de inserir Nome, CPF, idade e mensagem separados por vírgula.")
                    continue

                nome = data_parts[0].strip()
                cpf = data_parts[1].strip()

                try:
                    idade = int(data_parts[2].strip())
                except ValueError:
                    print("Idade deve ser um número inteiro.")
                    continue

                mensagem = data_parts[3].strip()

                json_message = {
                    "Nome": nome,
                    "CPF": cpf,
                    "idade": idade,
                    "mensagem": mensagem
                }

                client_socket.sendall(json.dumps(json_message).encode() + b"\n")

    except Exception as e:
        print(e)

if __name__ == "__main__":
    main()
